### DISCUSSION TITLE
[Kavita] Idea / Feature Submission Support Server-Based Databases for Better Flexibility in Self-Hosting

### DISCUSSION THREAD

**Discussion by unlucio** [Kareadita/Kavita#3431] [Ideas]
### Idea Description

Hi everyone,
Thanks for making this project—it’s been great to use! I do have a suggestion regarding the use of SQLite. While it’s fine for simple setups, it struggles when the database file is on a network share. This can be a problem for those of us using NAS setups or deploying apps on Docker Swarm or k3s clusters.
Using a server-based database like MySQL or PostgreSQL would fit better with these environments. If the project uses an ORM (like Entity Framework), adding support for configurable database connections shouldn’t be too difficult. It would give users the flexibility to choose what works best for their setup while keeping SQLite for simpler use cases.
This change would make the project more convenient for homelab enthusiasts running apps across old hardware or in containerized environments. Let me know if this sounds feasible—I’d be happy to test it out or help however I can!

Cheers

### Idea Category

User Experience

### Duration of Using Kavita

I'd love to use it, but SQLite-only is too much of a headache for self hosting

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
I have no plans to support anything other than SQLite.

**Comment by DieselTech** [comment]
To be clear, sqlite can run over a network. It just can't do it over your "normal" network protocols like SMB or NFS. It needs a proper filesystem that has locking in order to do so, like iSCSI. It is just easier from a support perspective to say it doesn't run over the network than for me to explain what a network filesystem is to most people. 

Also it kind of goes against what you are saying here:

> more convenient for homelab enthusiasts

I'd argue running a full thick SQL server isn't convenient for people who just want to host a few services.
